### DISCUSSION TITLE
Cannot read property 'data' of undefined

### DISCUSSION THREAD

**Discussion by Rajesh-Kodandam** [cypress-io/cypress#16987] [Cypress Cloud]
I'm working on cypress: 7.4.0

describe('verif', () => {
before (function() {
cy.fixture('example').then(function(data){
this.data = data;
});
});
It('valida', () => {
cy.log('');
});
})

Not sure the reason I have this error while trying to use the data from fixture "Cannot read property 'data' of undefined"

Your advice much appreciated.

**Comment by tit** [comment]
If you what to use fixtures from before block, you must use function callback, not arrow function () => {} in the it block.

https://docs.cypress.io/api/commands/fixture#this-context

```
describe('verif', () => {
  before (function() {
    cy.fixture('example').then(function(data){
      this.data = data;
    });
  });
  It('valida', function () {
    cy.log('');
  });
})
```

**Reply by Rajesh-Kodandam** [reply]
Thank you so much for your valuable inputs and it worked.
